In this step exploratory data analysis is performed on the dataset. The dataset is loaded and explored to get a better understanding of the data. Possible data types are noted and data is split into train and test set.

In [1]:
# Importing the libraries
import pandas as pd
from matplotlib import pyplot as plt
import numpy as np
%matplotlib inline

In [6]:
# load the dataset
full_df = pd.read_csv('../dataset/cell2celltrain.csv')
full_df.head()

CustomerID Churn  MonthlyRevenue  MonthlyMinutes  TotalRecurringCharge  \
0     3000002   Yes           24.00           219.0                  22.0   
1     3000010   Yes           16.99            10.0                  17.0   
2     3000014    No           38.00             8.0                  38.0   
3     3000022    No           82.28          1312.0                  75.0   
4     3000026   Yes           17.14             0.0                  17.0   

   DirectorAssistedCalls  OverageMinutes  RoamingCalls  PercChangeMinutes  \
0                   0.25             0.0           0.0             -157.0   
1                   0.00             0.0           0.0               -4.0   
2                   0.00             0.0           0.0               -2.0   
3                   1.24             0.0           0.0              157.0   
4                   0.00             0.0           0.0                0.0   

   PercChangeRevenues  ...  ReferralsMadeBySubscriber  IncomeGroup  \
0               -19.0  ...                          0            4   
1                 0.0  ...                          0            5   
2                 0.0  ...                          0            6   
3                 8.1  ...                          0            6   
4                -0.2  ...                          0            9   

   OwnsMotorcycle  AdjustmentsToCreditRating  HandsetPrice  \
0              No                          0            30   
1              No                          0            30   
2              No                          0       Unknown   
3              No                          0            10   
4              No                          1            10   

   MadeCallToRetentionTeam  CreditRating  PrizmCode    Occupation  \
0                      Yes     1-Highest   Suburban  Professional   
1                       No      4-Medium   Suburban  Professional   
2                       No        3-Good       Town        Crafts   
3                       No      4-Medium      Other         Other   
4                       No     1-Highest      Other  Professional   

   MaritalStatus  
0             No  
1            Yes  
2            Yes  
3             No  
4            Yes  

[5 rows x 58 columns]

In [7]:
full_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51047 entries, 0 to 51046
Data columns (total 58 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   CustomerID                 51047 non-null  int64  
 1   Churn                      51047 non-null  object 
 2   MonthlyRevenue             50891 non-null  float64
 3   MonthlyMinutes             50891 non-null  float64
 4   TotalRecurringCharge       50891 non-null  float64
 5   DirectorAssistedCalls      50891 non-null  float64
 6   OverageMinutes             50891 non-null  float64
 7   RoamingCalls               50891 non-null  float64
 8   PercChangeMinutes          50680 non-null  float64
 9   PercChangeRevenues         50680 non-null  float64
 10  DroppedCalls               51047 non-null  float64
 11  BlockedCalls               51047 non-null  float64
 12  UnansweredCalls            51047 non-null  float64
 13  CustomerCareCalls          51047 non-null  flo

In [8]:
full_df.isna().sum()

CustomerID                     0
Churn                          0
MonthlyRevenue               156
MonthlyMinutes               156
TotalRecurringCharge         156
DirectorAssistedCalls        156
OverageMinutes               156
RoamingCalls                 156
PercChangeMinutes            367
PercChangeRevenues           367
DroppedCalls                   0
BlockedCalls                   0
UnansweredCalls                0
CustomerCareCalls              0
ThreewayCalls                  0
ReceivedCalls                  0
OutboundCalls                  0
InboundCalls                   0
PeakCallsInOut                 0
OffPeakCallsInOut              0
DroppedBlockedCalls            0
CallForwardingCalls            0
CallWaitingCalls               0
MonthsInService                0
UniqueSubs                     0
ActiveSubs                     0
ServiceArea                   24
Handsets                       1
HandsetModels                  1
CurrentEquipmentDays           1
AgeHH1    

The dataset has 51047 rows and 58 columns. Multiple of those columns seem to be of object type. Out of 58 columns only a handful of them contain missing values. Maximum number of missing values of a column is 909 from AgeHH1, AgeHH2.

In [13]:
# explore the object type columns
def print_unique_col_values_counts (df):
    for column in df:
        if df[column].dtypes=="object":
            print(f'{column}: {df[column].value_counts()}')

In [14]:
print_unique_col_values_counts(full_df)

Churn: No     36336
Yes    14711
Name: Churn, dtype: int64
ServiceArea: NYCBRO917    1684
HOUHOU281    1510
DALDAL214    1498
NYCMAN917    1182
APCFCH703     783
             ... 
OHIRAV330       1
OMANEW515       1
KCYHES316       1
AIRGEO843       1
NCRDNN910       1
Name: ServiceArea, Length: 747, dtype: int64
ChildrenInHH: No     38679
Yes    12368
Name: ChildrenInHH, dtype: int64
HandsetRefurbished: No     43956
Yes     7091
Name: HandsetRefurbished, dtype: int64
HandsetWebCapable: Yes    46046
No      5001
Name: HandsetWebCapable, dtype: int64
TruckOwner: No     41524
Yes     9523
Name: TruckOwner, dtype: int64
RVOwner: No     46894
Yes     4153
Name: RVOwner, dtype: int64
Homeownership: Known      33987
Unknown    17060
Name: Homeownership, dtype: int64
BuysViaMailOrder: No     32596
Yes    18451
Name: BuysViaMailOrder, dtype: int64
RespondsToMailOffers: No     31821
Yes    19226
Name: RespondsToMailOffers, dtype: int64
OptOutMailings: No     50295
Yes      752
Name: OptOutMaili

It can be observed that the object columns are categorical in nature.The missing values are represented by 'Unknown' in the dataset. ategorical features are converted to numerical features using one hot encoding later into the process.